backend/fms_core/viewsets/_fetch_data.py

```diff
@@ -333,6 +333,87 @@ def fetch_export_data(self, ids: List[int] =[]) -> Tuple[List, int]:
 
         return serialized_data
 
+    def fetch_export_metadata(self, ids: List[int] =[]) -> Tuple[List, int]:
+        """
+        Function to retrieve the sample metadata to export. 
+
+        Args:
+            self: base_class(viewset) + fetch_data
+            ids: List of ids to select specific samples. Defaults to [].
+
+        Returns:
+            Returns a list of serialized sample metadata dictionary
+        """
+
+        super().fetch_export_data(ids) # Initialize queryset by calling base abstract function
+
+        self.queryset = self.queryset.values(
+            'id',
+            'name',
+            'container__kind',
+            'container__name',
+            'container__barcode',
+            'coordinates',
+            'container__location__barcode',
+            'container__coordinates',
+            'count_derived_samples',
+            'first_derived_sample',
+        )
+        samples = {s["id"]: s for s in self.queryset}
+
+        first_derived_sample_ids = self.queryset.values_list("first_derived_sample", flat=True)
+        derived_sample_values_queryset = (
+            DerivedSample.objects
+            .filter(id__in=first_derived_sample_ids)
+            .annotate(is_library=ExpressionWrapper(Q(library__isnull=False), output_field=BooleanField()))
+            .values(
+                'id',
+                'biosample__id',
+                'biosample__alias',
+                'project__name',
+                'is_library',
+            )
+        )
+        derived_samples = {ds["id"]: ds for ds in derived_sample_values_queryset}
+
+        biosample_ids = derived_sample_values_queryset.values_list('biosample__id', flat=True)
+        metadata_queryset = SampleMetadata.objects.filter(biosample_id__in=biosample_ids)
+        metadata_obj = metadata_queryset.values('biosample','name', 'value')
+
+        metadata_per_biosample = {}
+        for metadata in metadata_obj:
+            biosample = metadata['biosample']
+            if biosample in metadata_per_biosample.keys():
+                metadata_per_biosample[biosample].append(metadata)
+            else:
+                metadata_per_biosample[biosample] = [metadata] 
+
+        serialized_data = []
+        if not samples:
+            serialized_data.append({}) # Allow the returned csv file to be named instead of random name.
+        for sample in samples.values():
+            derived_sample = derived_samples[sample["first_derived_sample"]]
+            is_library = derived_sample["is_library"]
+            is_pool = sample["count_derived_samples"] > 1
+            metadata = metadata_per_biosample[derived_sample["biosample__id"]]
```

Comment:
This will be wrong for pools ....

Reply:
Since there is nothing to be build around this we can "break down" the sample into derived samples ... each line would be a derived sample/ biosample... with their matching metadata.